R/count_cumulative.R

```diff
@@ -144,10 +145,80 @@ s_count_cumulative <- function(x,
 #' * `a_count_cumulative()` returns the corresponding list with formatted [rtables::CellValue()].
 #'
 #' @keywords internal
-a_count_cumulative <- make_afun(
-  s_count_cumulative,
-  .formats = c(count_fraction = format_count_fraction)
-)
+a_count_cumulative <- function(x,
+                               ...,
+                               .stats = NULL,
+                               .stat_names = NULL,
+                               .formats = NULL,
+                               .labels = NULL,
+                               .indent_mods = NULL) {
+  dots_extra_args <- list(...)
+
+  # Check if there are user-defined functions
+  default_and_custom_stats_list <- .split_std_from_custom_stats(.stats)
+  .stats <- default_and_custom_stats_list$all_stats
+  custom_stat_functions <- default_and_custom_stats_list$custom_stats
+
+  # Adding automatically extra parameters to the statistic function (see ?rtables::additional_fun_params)
+  extra_afun_params <- retrieve_extra_afun_params(
+    names(dots_extra_args$.additional_fun_parameters)
+  )
+  dots_extra_args$.additional_fun_parameters <- NULL # After extraction we do not need them anymore
+
+  # Main statistical functions application
+  x_stats <- .apply_stat_functions(
+    default_stat_fnc = s_count_cumulative,
+    custom_stat_fnc_list = custom_stat_functions,
+    args_list = c(
+      x = list(x),
+      extra_afun_params,
+      dots_extra_args
+    )
+  )
+
+  # Fill in with stats defaults if needed
+  .stats <- get_stats("count_cumulative",
+    stats_in = .stats,
+    custom_stats_in = names(custom_stat_functions)
+  )
+
+  x_stats <- x_stats[.stats]
+  levels_per_stats <- lapply(x_stats, names)
+
+  # Fill in formats/indents/labels with custom input and defaults
+  .formats <- get_formats_from_stats(.stats, .formats, levels_per_stats)
+  .indent_mods <- get_indents_from_stats(.stats, .indent_mods, levels_per_stats)
+  if (is.null(.labels)) {
+    .labels <- sapply(.unlist_keep_nulls(x_stats), attr, "label")
+    .labels <- .labels[nzchar(.labels) & !sapply(.labels, is.null) & !is.na(.labels)]
+  }
+  .labels <- get_labels_from_stats(.stats, .labels, levels_per_stats)
```

Comment:
Again can use these as defaults for greater customizability.
```suggestion
  labels_def <- lapply(x_stats[["count_fraction"]], attr, "label")
  .labels <- get_labels_from_stats(.stats, .labels, levels_per_stats, labels_def)
```
```

Reply:
The issue is that sometimes it is `""` the label attribute.

> get_labels_from_stats("count", NULL, levels_per_stats = list("count" = c("a", "b")), tern_defaults = list("count" = ""))
$count.a
[1] ""

$count.b
[1] ""

> get_labels_from_stats("count")
$count
[1] "count"

Reply:
Should we take only one x_stats out? I know the default is only one here but maybe it is a bit too stringent?

Reply:
I would assume that there's a reason why the label is set to `""` sometimes - if not, we should probably just remove or change that default. What do you mean by "take only one x_stats out"? Also, isn't `count_fraction` the only statistic returned by `s_count_cumulative`?

Reply:
If you wan to add your custom stats here you cannot due to the x_stats[["count_fraction"]]. It is the only stat returned at the moment, I am not sure it will always be like that!

Reply:
I tried to make it more general. See other comments ;)